mmaction/core/evaluation/average_precision.py

```diff
@@ -0,0 +1,315 @@
+import os.path as osp
+import pickle
+from collections import defaultdict
+
+import numpy as np
+
+from .bbox_overlaps import iou2d, spatio_temporal_iou3d, spatio_temporal_nms3d
+
+
+def pr_to_ap(precision_recall):
+    """Compute AP given precision-recall.
+
+    Args:
+        precision_recall (np.ndarray): precision_recall is an Nx2 array with
+            first row being precision and second row being recall.
+
+    Returns:
+        np.ndarray: The result of average precision.
+    """
+
+    pr_diff = precision_recall[1:, 1] - precision_recall[:-1, 1]
```

Comment:
pr?

Reply:
you mean `precision_recall` -> `pr`, or `pr` -> `precision_recall` ?

Reply:
![image](https://user-images.githubusercontent.com/9464825/100612009-3800b180-334d-11eb-9c37-bc3af3e69e99.png)

okay, it does not matter

Reply:
recall diff and precision sum?